app/assets/javascripts/discourse/app/components/modal/create-invite.gjs

```diff
@@ -0,0 +1,503 @@
+              >
+                <field.Select as |select|>
+                  {{#each this.expireAfterOptions as |option|}}
+                    <select.Option
+                      @value={{option.value}}
+                    >{{option.text}}</select.Option>
+                  {{/each}}
+                </field.Select>
+              </form.Field>
+            {{/if}}
+
+            {{#if this.canArriveAtTopic}}
+              <form.Field
+                @name="inviteToTopic"
+                @title={{i18n "user.invited.invite.invite_to_topic"}}
+                @format="large"
+                as |field|
+              >
+                <field.Custom>
+                  <TopicChooser
+                    @value={{field.value}}
+                    @content={{this.topics}}
+                    @onChange={{fn this.onChangeTopic field.set}}
+                    @options={{hash additionalFilters="status:public"}}
+                  />
+                </field.Custom>
+              </form.Field>
+            {{/if}}
+
+            {{#if this.canInviteToGroup}}
+              <form.Field
+                @name="inviteToGroups"
+                @title={{i18n "user.invited.invite.add_to_groups"}}
+                @format="large"
+                as |field|
+              >
+                <field.Custom>
+                  <GroupChooser
+                    @content={{this.allGroups}}
+                    @value={{field.value}}
+                    @labelProperty="name"
+                    @onChange={{field.set}}
+                  />
+                </field.Custom>
+              </form.Field>
+            {{/if}}
+
+            {{#if (emailValid transientData.restrictTo)}}
+              <form.Field
+                @name="customMessage"
+                @title={{i18n "user.invited.invite.custom_message"}}
+                @format="full"
+                as |field|
+              >
+                <field.Textarea
+                  height={{100}}
+                  placeholder={{i18n
+                    "user.invited.invite.custom_message_placeholder"
+                  }}
+                />
+              </form.Field>
+            {{/if}}
+          </Form>
+        {{/if}}
+      </:body>
+      <:footer>
+        {{#if this.simpleMode}}
+          <DButton
+            @label="user.invited.invite.create_link"
+            @action={{this.createLink}}
+            @disabled={{this.saving}}
+            class="btn-primary save-invite"
+          />
+        {{else}}
+          <DButton
+            @label="user.invited.invite.save_invite"
+            @action={{this.saveInvite}}
+            @disabled={{this.saving}}
+            class="btn-primary save-invite"
+          />
+        {{/if}}
+        <DButton
+          @label="user.invited.invite.cancel"
+          @action={{this.cancel}}
+          class="btn-transparent cancel-button"
+        />
+      </:footer>
+    </DModal>
+  </template>
+}
+
+const InviteModalAlert = <template>
```

Comment:
Also probably move this to its own file. We want to avoid multiple components per file generally...I think.

Reply:
I think multiple components per file is ok if they're very closely related, it's even mentioned in Ember docs https://guides.emberjs.com/release/components/template-tag-format/#toc_multiple-components-per-file.

The 2 non-exported components here are meant be "private", i.e., they only make sense within the main/public component in this file, so I think it'd be a little bit icky to have them in their own files when nothing else can/should use them.

Reply:
Interesting, I didn't realise the docs said that. Fine to keep them together then if they are "private" components :+1: